### PR TITLE
Cleanup CI

### PR DESCRIPTION
**Pull request by FFroehlich**
Move more functionality to GitHub Actions, fix coverage report

### CODE REVIEW

File: .github/workflows/deploy_gh_pages.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: Doxygen
```
Comment (dweindl) [inline .github/workflows/deploy_gh_pages.yml:1]:
```suggestion
name: Doxygen / gh-pages
```

File: .github/workflows/test-petab-test-suite.yml [hunk 2]
```diff
@@ -28,35 +28,33 @@ jobs:
...
-    - name: Create AMICI sdist
+    - name: Install python package
       run: |
-        cd python/sdist \
-          && check_time.sh create_sdist /usr/bin/python3 setup.py sdist
-    - name: Install AMICI sdist
-      run: |
-        AMICI_PARALLEL_COMPILE=2 check_time.sh \
-            install_sdist pip3 install -v --user \
-            $(ls -t python/sdist/dist/amici-*.tar.gz | head -1)[petab,pysb]
```
Comment (dweindl) [inline .github/workflows/test-petab-test-suite.yml:50]: Do we still have the installation from sdist somewhere? Should be tested somewhere.

File: .github/workflows/test_python_ver_matrix.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+          && pip3 install --upgrade --user setuptools
+          
+
+    # install AMICI
+    - name: Build BNGL
+      run: |
+        scripts/buildBNGL.sh
+    - name: Install python package
+      run: |
+        scripts/installAmiciSource.sh
```
Comment (dweindl) [inline .github/workflows/test_python_ver_matrix.yml:54]: Maybe use sdist here, as that is the way how most users will use it?

File: include/amici/vector.h [hunk 4]
```diff
@@ -61,6 +61,10 @@ class AmiVector {
             N_VMake_Serial(static_cast<long int>(vold.vec_.size()), vec_.data());
     }
 
+    /**
+     * @brief move constructor
+     * @param other vector from which the data will be moved
```
Comment (dweindl) [inline include/amici/vector.h:66]: Does doxygen complain about those? It doesn't really add value to document the obvious...

File: .github/workflows/test_python_ver_matrix.yml (hunk 3 above)
Comment (FFroehlich) [reply .github/workflows/test_python_ver_matrix.yml:54]: Had that initially, but having notebooks run outside a virtualenv was a major pain

File: include/amici/vector.h (hunk 4 above)
Comment (FFroehlich) [reply include/amici/vector.h:66]: yep

File: .github/workflows/test-petab-test-suite.yml (hunk 2 above)
Comment (FFroehlich) [reply .github/workflows/test-petab-test-suite.yml:50]: still in use in test-large-model.yaml

**Pull request merged**